bigtable/client/cell.h

```diff
@@ -0,0 +1,99 @@
+// Copyright 2017 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef BIGTABLE_CLIENT_CELL_H_
+#define BIGTABLE_CLIENT_CELL_H_
+
+#include <absl/base/thread_annotations.h>
+#include <absl/strings/string_view.h>
+#include <bigtable/client/version.h>
+#include <mutex>
+#include <vector>
+
+namespace bigtable {
+inline namespace BIGTABLE_CLIENT_NS {
+/**
+ * The in-memory representation of a Bigtable cell.
+ *
+ * Bigtable stores data in rows, indexes by row keys.  Each row may contain
+ * multiple column families, each column family might contain multiple columns,
+ * and each column has multiple cells indexed by timestamp.  Notice that the
+ * storage is sparse, column families, columns, and timestamps might contain
+ * zero cells.
+ *
+ * The Cell class owns all its data. Its string accessors return
+ * absl::string_view, which is a lightweight pointer to the string contained in
+ * Cell and does not actually hold any data. If those values are needed beyond
+ * the lifetime of the Cell object itself, they need to be copied.
+ */
+class Cell {
+ public:
+  /// Create a Cell and fill it with data.
+  Cell(std::string row_key, std::string family_name,
+       std::string column_qualifier, int64_t timestamp,
+       std::vector<std::string> chunks, std::vector<std::string> labels)
+      : row_key_(std::move(row_key)),
+        family_name_(std::move(family_name)),
+        column_qualifier_(std::move(column_qualifier)),
+        timestamp_(timestamp),
+        chunks_(std::move(chunks)),
+        labels_(std::move(labels)) {}
+
+  /// Return the row key this cell belongs to. The returned value is not valid
+  /// after this object dies.
+  absl::string_view row_key() const { return row_key_; }
+  /// Return the family this cell belongs to. The returned value is not valid
```

Comment:
Insert blank like please. Also before the other function declaration / definitions.

Reply:
done